plugin/src/main/kotlin/com/android/gradle/replicator/ResourceDataGathering.kt

```diff
@@ -0,0 +1,134 @@
+    return AndroidSizeMattersResourceProperties(
+            qualifiers = qualifiers,
+            extension = extension,
+            quantity = resourceFiles.size,
+            fileSizes = mutableListOf<Long>().apply {
+                resourceFiles.forEach {
+                    this.add(it.length())
+                }
+            }
+    )
+}
+
+fun getDefaultResourceData(qualifiers: String, extension: String, resourceFiles: Set<File>): AbstractAndroidResourceProperties {
+    return AndroidDefaultResourceProperties(
+            qualifiers = qualifiers,
+            extension = extension,
+            quantity = resourceFiles.size
+    )
+}
+
+
+fun parseValuesFile(valuesFile: File): ValuesMap {
+
+    val valuesMap = ValuesMap()
+
+    val handler: DefaultHandler = object : DefaultHandler() {
+        private var myDepth = 0
+        private var myLocator: Locator? = null
+        private var currentItemCount = 0
+        override fun setDocumentLocator(locator: Locator) {
+            myLocator = locator
+        }
+        override fun startElement(
+            uri: String, localName: String, qName: String, attributes: Attributes
+        ) {
+            myDepth++
+
+            if (myDepth == 3 && qName == "item") {
+                currentItemCount++
+            }
+
+            if (myDepth == 2) {
+                when (qName) {
+                    "string" -> { valuesMap.stringCount++ }
+                    "int" -> { valuesMap.intCount++ }
+                    "bool" -> { valuesMap.boolCount++  }
+                    "color" -> { valuesMap.colorCount++  }
+                    "dimen" -> { valuesMap.dimenCount++  }
+                    "item" -> { if (attributes.getValue("name") == "id") { valuesMap.idCount++ } }
+                    "integer-array" -> {} // Only add on close
+                    "array" -> {} // Only add on close
+                    "style" -> {} // Only add on close
+                    else -> {} // Ignore
+                }
+            }
+        }
+
+        override fun endElement(uri: String, localName: String, qName: String) {
+            if (myDepth == 2) {
+                when (qName) {
+                    "integer-array" -> {
+                        valuesMap.integerArrayCount.add(currentItemCount)
+                    }
+                    "array" -> {
+                        valuesMap.arrayCount.add(currentItemCount)
+                    }
+                    "style" -> {
+                        valuesMap.styleCount.add(currentItemCount)
+                    }
+                    else -> { }
+                }
+                currentItemCount = 0
+            }
+            myDepth--
+        }
+    }
+
+    val factory = SAXParserFactory.newInstance()
+    // Parse the input
+    XmlUtils.configureSaxFactory(factory, false, false)
+    val saxParser = XmlUtils.createSaxParser(factory)
+    saxParser.parse(InputSource(StringReader(valuesFile.readText())), handler)
```

Comment:
this code need to be super resilient. so for instance, here you should handle the case the input file is incorrect, log it and continue if possible...